PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/AbstractAzureTextInput.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+package com.microsoft.azure.toolkit.intellij.common;
+
+import com.google.common.collect.ImmutableMap;
+import com.intellij.icons.AllIcons;
+import com.intellij.ui.AnimatedIcon;
+import com.intellij.ui.components.fields.ExtendableTextField;
+import com.microsoft.azure.toolkit.lib.common.form.AzureValidationInfo;
+import com.microsoft.azure.toolkit.lib.common.utils.Debouncer;
+import com.microsoft.azure.toolkit.lib.common.utils.TailingDebouncer;
+
+import javax.annotation.Nullable;
+import javax.swing.*;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+
+public class AbstractAzureTextInput<T> extends ExtendableTextField
+    implements AzureFormInputComponent<T>, TextDocumentListenerAdapter {
+    protected static final int DEBOUNCE_DELAY = 500;
+    private final Debouncer debouncer;
+    private static final Extension VALIDATING = Extension.create(AnimatedIcon.Default.INSTANCE, "Validating...", null);
+    private static final Extension SUCCESS = Extension.create(AllIcons.General.InspectionsOK, "Validation passed.", null);
+    private static final Map<AzureValidationInfo.Type, Function<AzureValidationInfo, Extension>> extensions = ImmutableMap.of(
+        AzureValidationInfo.Type.PENDING, (i) -> VALIDATING,
+        AzureValidationInfo.Type.SUCCESS, (i) -> SUCCESS,
+        AzureValidationInfo.Type.ERROR, (i) -> Extension.create(AllIcons.General.BalloonError, i.getMessage(), null),
+        AzureValidationInfo.Type.WARNING, (i) -> Extension.create(AllIcons.General.BalloonWarning, i.getMessage(), null)
+    );
+
+    public AbstractAzureTextInput() {
+        this(null);
+    }
+
+    public AbstractAzureTextInput(@Nullable JTextField comp) {
```

Comment:
is there any cases for null component?

Reply:
the non-arg constructor. null will default to the `getInputComponent()`